client/src/partials/cash/modals/transfer.modal.js

```diff
@@ -0,0 +1,110 @@
+angular.module('bhima.controllers')
+.controller('CashTransferModalController', CashTransferModalController);
+
+CashTransferModalController.$inject = [
+  '$uibModalInstance', 'SessionService', 'CurrencyService', 'VoucherService', 'CashboxService', 'AccountService', 'cashBox', 'uuid'
+];
+
+/**
+ * @module cash/modals/CashTransferModalController
+ *
+ * @description This controller is responsible transfering money between auxillary cash and a virement account
+*/
+
+function CashTransferModalController(ModalInstance, sessionService, currencyService, voucherService, cashBoxService, accountService, cashBox, uuid) {
+  var vm = this; 
+
+  /** Attaching service to the scope **/
+  vm.cashBox = cashBox;
+  vm.currencyService = currencyService; 
+  vm.cashBoxService = cashBoxService;
+
+  /** init success to false**/
+  vm.success = false;
+
+  /**
+  * @function submit
+  * @param {boolean} invalid it is true if the form is valid
+  * @description for submitting a dialog content
+  **/
+  function submit (invalid){
+    if (invalid) { return; }
+    var voucher = initializeVoucher();
+    var voucher_item = initializeVoucherItem(voucher.uuid);
+    var record = { voucher : voucher, voucher_item : voucher_item };
+    voucherService.create(record)      
+    .then(function (res){
+      vm.success = true;
+    });
+  }
+
+  function cancel (){
+    ModalInstance.dismiss();
+  }
+
+  /**
+  * This methode is responsible to generate a description for the voucher operation
+  * @private
+  **/
+  function generateDescription (){
+    return ['Transfer voucher', new Date().toISOString().slice(0, 10)].join('/');
+  }
+
+  /**
+  * This methode is responsible to create a voucher object and it back
+  * @private
+  **/
+  function initializeVoucher (){
```

Comment:
This code still belongs in either a client service or the server, not the controller.